### DISCUSSION TITLE
How to debug task defined in sbt using dap?

### DISCUSSION THREAD

**Discussion by gogamid** [scalameta/nvim-metals#600] [Q&A] [answered]
I have basically run configurations defined in build.sbt file, they run scala code with different arguments per each run config. Example:

```sbt
val prems = inputKey[Unit]("premium small trace")
prems := (Compile / run).fullInput(getArguments("LF481773_1Min-Part0.ttl", premiumPath)).evaluated
```
to run it i execute this command
`sbt prems`

**What should i do to be able to debug a task defined in build.sbt?**

I have seen this configuration option in nvim config, but i do not want my run configuration to be defined or saved outside of project folder. 
 ```lua
     {
        type = "scala",
        request = "launch",
        name = "run-someip",
        metals = {
          mainClass = "Main",
          runType = "run",
          buildTarget = "root",
          --args = { "firstArg", "secondArg", "thirdArg" }, -- here just as an example
        },
      },
```
Can you advise me on this?

**Comment by ckipp01** [comment]
Hey @gogamid, I'm not actually sure if these are processed or not, but one thing to actually try would be to use sbt as your build server instead of Bloop. You can use `:MetalsSwitchBsp`, and then choose `sbt`. Could you try and report back to see if the behavior is the same?

@adpi2 do you by chance know if this would work with sbt BSP?

**Reply by adpi2** [reply, marked as answer]
Unfortunately it would not work. There is no way currently to configure the debugger outside of the nvim config (or VSCode launch configuration). This is not great and I wonder how we could improve this. Would it make sense to use using directives to embed the debug configuration into Scala files? (maybe I should open a discussion in Metals about this)

**Comment by doctorstal** [comment]
Any chance the situation changed in recent years?

My workflow depends on sbt tasks and I thought that is common among scala devs.
I.e. that's most common way of running stuff like Play app.

I thought there should be a way of running sbt and then attaching the debugger to it, right?
